pkg/runtime/wfengine/backends/actors/workflow_actor.go

```diff
@@ -672,7 +682,7 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, reminder actors.Intern
 
 	err = wf.saveInternalState(ctx, state)
 	if err != nil {
-		return err
+		return runCompletedTrue, err
```

Comment:
Shouldn't this be `runCompletedFalse`? saving state is retriable?

Reply:
Yes, I think it probably should to but the _current_ implementation treats this as a terminal error and didn't want to change behaviour in this PR.

https://github.com/dapr/dapr/blob/ca411ccd824f60a2d7286ff39c9c7f4d3bd64102/pkg/runtime/wfengine/backends/actors/workflow_actor.go#L166

Reply:
Ok. lets leave this for now, but this is a question for @cgillum also.

Reply:
I unfortunately don't remember this clearly. It seems like it should be a recoverable error. Perhaps I was worried about the cases where it's not recoverable, such as when you run into certain state store limitations around payload sizes. Happy to revisit this later.